In [1]:

import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns 
%matplotlib inline 

In [2]:
df = pd.read_csv("/storage/emulated/0/Mycsvfile/FoodBalanceSheets.csv", encoding = "latin-1")
df

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [3]:
df.shape

(60943, 12)

Our Dataset contains :
60943 rows and 12 columns. 

In [4]:
df.columns.tolist () 

['Area Code',
 'Area',
 'Item Code',
 'Item',
 'Element Code',
 'Element',
 'Unit',
 'Y2014',
 'Y2015',
 'Y2016',
 'Y2017',
 'Y2018']

In [5]:


nul=df["Y2014"] .isnull().sum()
print("Number if missing value in Y2014 is", nul) 

Number if missing value in Y2014 is 1589


In [6]:
total_nul = sum (df.isna().sum())
print("total missing values in our Dataset is ", total_nul) 
Y2014_nul_percent = (nul/total_nul) * 100
print ("Year 2014 missing values percentage is ", round(Y2014_nul_percent, 3) ) 




total missing values in our Dataset is  7614
Year 2014 missing values percentage is  20.869


In [7]:
group1 = df[["Area", "Y2017"]]. groupby ("Area").sum() 
group1.sort_values (by="Y2017", ascending = False  ) 


Y2017
Area                                   
Nigeria                      1483268.23
Egypt                         866379.92
South Africa                  517590.54
Ethiopia                      448683.76
Morocco                       388495.36
Ghana                         337599.06
Algeria                       325644.27
United Republic of Tanzania   322616.85
Kenya                         264660.66
Sudan                         239931.92
Cameroon                      232030.43
Angola                        229159.57
Côte d'Ivoire                 224599.01
Uganda                        213950.38
Malawi                        181098.71
Mozambique                    161407.98
Mauritania                    156665.46
Mali                          149928.33
Madagascar                    131197.73
Niger                         126707.58
Benin                         124771.22
Tunisia                       124167.20
Zambia                        103223.77
Burkina Faso                  101855.07
Guinea                         98138.87
Senegal                        95681.15
Zimbabwe                       75919.34
Rwanda                         73663.69
Chad                           71594.68
Sierra Leone                   55311.33
Eswatini                       54343.33
Mauritius                      51114.83
Togo                           49841.88
Congo                          41181.68
Central African Republic       29937.00
Namibia                        29874.89
Liberia                        29342.20
Gabon                          27979.64
Gambia                         23154.18
Djibouti                       22729.91
Botswana                       22101.30
Lesotho                        21267.96
Guinea-Bissau                  19102.77
Cabo Verde                     14650.74
Sao Tome and Principe          12662.63
Seychelles                       442.34
Comoros                           59.84
Sudan (former)                     0.00
Ethiopia PDR                       0.00

From the above code we can see that Nigeria has the highest sum in 2017 

In [8]:
cor = df[["Element Code", "Y2014", "Y2015", "Y2016", "Y2017", "Y2018" ]].corr().sort_values(by = "Element Code"    ) 
cor 



Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
Y2016             0.023444  0.996081  0.995739  1.000000  0.992785  0.992757
Y2015             0.023889  0.994647  1.000000  0.995739  0.988048  0.988208
Y2017             0.024254  0.995230  0.988048  0.992785  1.000000  0.998103
Y2018             0.024279  0.994872  0.988208  0.992757  0.998103  1.000000
Y2014             0.024457  1.000000  0.994647  0.996081  0.995230  0.994872
Element Code      1.000000  0.024457  0.023889  0.023444  0.024254  0.024279

Form this code Y2016 has the lowest correlation with Element Code column 

In [9]:
std = df["Y2017"].std() 
mean = df["Y2017"].mean() 
print("The Standard deviation for the year 2017 is ",round( std, 2))  
print("The mean is ", round(mean, 2))  



The Standard deviation for the year 2017 is  1671.86
The mean is  140.92


In [10]:
group2 = df[["Element", "Y2014", "Y2015", "Y2016", "Y2017", "Y2018"]].groupby("Element").sum() 
stock = group2.sort_values(by="Element") 
print (stock.loc ["Stock Variation"]) 
stock.loc["Stock Variation" ].max() 



Y2014    58749.83
Y2015    34910.99
Y2016    33140.12
Y2017    54316.91
Y2018    20577.91
Name: Stock Variation, dtype: float64


58749.83

From this code we can conclude that year 2014 has the highest Stock Variation 

In [11]:
print ("The total number of unique countries in our Dataset is", df["Area"].nunique()) 


The total number of unique countries in our Dataset is 49


In [55]:
group3 = df[["Area", "Y2017"]].groupby ("Area"). sum() 
group3.sort_values (by = "Y2017", ascending = True  ) 



Y2017
Area                                   
Sudan (former)                     0.00
Ethiopia PDR                       0.00
Comoros                           59.84
Seychelles                       442.34
Sao Tome and Principe          12662.63
Cabo Verde                     14650.74
Guinea-Bissau                  19102.77
Lesotho                        21267.96
Botswana                       22101.30
Djibouti                       22729.91
Gambia                         23154.18
Gabon                          27979.64
Liberia                        29342.20
Namibia                        29874.89
Central African Republic       29937.00
Congo                          41181.68
Togo                           49841.88
Mauritius                      51114.83
Eswatini                       54343.33
Sierra Leone                   55311.33
Chad                           71594.68
Rwanda                         73663.69
Zimbabwe                       75919.34
Senegal                        95681.15
Guinea                         98138.87
Burkina Faso                  101855.07
Zambia                        103223.77
Tunisia                       124167.20
Benin                         124771.22
Niger                         126707.58
Madagascar                    131197.73
Mali                          149928.33
Mauritania                    156665.46
Mozambique                    161407.98
Malawi                        181098.71
Uganda                        213950.38
Côte d'Ivoire                 224599.01
Angola                        229159.57
Cameroon                      232030.43
Sudan                         239931.92
Kenya                         264660.66
United Republic of Tanzania   322616.85
Algeria                       325644.27
Ghana                         337599.06
Morocco                       388495.36
Ethiopia                      448683.76
South Africa                  517590.54
Egypt                         866379.92
Nigeria                      1483268.23

In [12]:
group4 = df[["Element", "Area", "Y2015"]] .groupby(["Area", "Element"] ).sum () 
#madagascar = group4.loc[:,"Madagascar"] 
group4.loc["Madagascar"] 

Y2015
Element                                         
Domestic supply quantity                31214.98
Export Quantity                           494.75
Fat supply quantity (g/capita/day)         91.85
Feed                                     2070.22
Food                                    21120.65
Food supply (kcal/capita/day)            7685.00
Food supply quantity (kg/capita/yr)       871.59
Import Quantity                          1721.80
Losses                                   2176.00
Other uses (non-food)                    3957.12
Processing                               1803.00
Production                              29482.89
Protein supply quantity (g/capita/day)    173.05
Residuals                                -616.00
Seed                                      699.00
Stock Variation                          -505.00
Total Population - Both sexes           24234.00

The Quantity of proteins supply in Madagascar in 2015 is 173.05

In [13]:
group5 = df[["Item", "Y2015", "Y2018"]] .groupby("Item").sum()
print ( group5.loc["Wine"]) 

print("total Wine in 2015", 4251.81)
print("Total Wine in 2018", 4039.32)

Y2015    4251.81
Y2018    4039.32
Name: Wine, dtype: float64
total Wine in 2015 4251.81
Total Wine in 2018 4039.32


In [14]:
group6 = df[["Element", "Y2017"]].groupby ("Element"). sum()   
group6.loc["Processing"] 


Y2017    292836.0
Name: Processing, dtype: float64

In [15]:
df.duplicated().sum() 

0

In [16]:
df.dtypes.value_counts() 

float64    5
object     4
int64      3
dtype: int64

In [17]:
df.describe (). T

count         mean          std     min     25%      50%  \
Area Code     60943.0   134.265576    72.605709     4.0    74.0   136.00   
Item Code     60943.0  2687.176706   146.055739  2501.0  2562.0  2630.00   
Element Code  60943.0  3814.856456  2212.007033   511.0   684.0  5142.00   
Y2014         59354.0   134.196282  1567.663696 -1796.0     0.0     0.09   
Y2015         59395.0   135.235966  1603.403984 -3161.0     0.0     0.08   
Y2016         59408.0   136.555222  1640.007194 -3225.0     0.0     0.08   
Y2017         59437.0   140.917765  1671.862359 -1582.0     0.0     0.10   
Y2018         59507.0   143.758381  1710.782658 -3396.0     0.0     0.07   

                  75%       max  
Area Code      195.00     276.0  
Item Code     2775.00    2961.0  
Element Code  5511.00    5911.0  
Y2014            8.34  176405.0  
Y2015            8.46  181137.0  
Y2016            8.43  185960.0  
Y2017            9.00  190873.0  
Y2018            9.00  195875.0